src/AddAtomicMutex.cpp

```diff
@@ -0,0 +1,344 @@
+#include "AddAtomicMutex.h"
+#include "IREquality.h"
+#include "IRMutator.h"
+#include "IROperator.h"
+
+namespace Halide {
+namespace Internal {
+
+using std::set;
+using std::string;
+
+namespace {
+
+class FindLoad : public IRGraphVisitor {
+public:
+    using IRGraphVisitor::visit;
+
+    set<string> symbols;
+    bool found = false;
+
+    void visit(const Load *op) override {
+        if (symbols.find(op->name) != symbols.end()) {
+            found = true;
+        }
+        include(op->predicate);
+        include(op->index);
+    }
+
+    bool find_load(Expr e, const set<string> &symbols) {
```

Comment:
This should be a constructor so that the set can be taken by reference

Reply:
Done.